src/System.Linq.Expressions/src/System/Linq/Expressions/IndexExpression.cs

```diff
@@ -472,7 +472,7 @@ private static void ValidateAccessorArgumentTypes(MethodInfo method, ParameterIn
 
                     Type pType = pi.ParameterType;
                     if (pType.IsByRef) throw Error.AccessorsCannotHaveByRefArgs($"{nameof(indexes)}[{i}]");
-                    TypeUtils.ValidateType(pType, $"{nameof(indexes)}[{i}]");
+                    TypeUtils.ValidateType(pType, nameof(indexes));
```

Comment:
Having an index in the param name is good (easier to debug code and reduces head scratching in complex code)
On the other hand, you could argue that the argument is "paramName", so a string with index is actually not a paramName anymore. But that's small